DataManagementSystem/DB/FileCatalogComponents/DirectoryMetadata.py

```diff
@@ -24,25 +25,52 @@ def setDatabase(self, database):
 #  Manage Metadata fields
 #
 
+  def getMetaName(self, meta, credDict):
+    """
+    Return a metadata name based on client supplied meta name and client credentials
+    For the base class it just returns the name passed in.
+    This method is meant to be overwritten by derived classes.
+
+    :param meta:  meta name
+    :param credDict: client credentials
+    :return: meta name
+    """
+
+    return meta
+
+  def getMetaNameSuffix(self, credDict):
+    """
+    Get meta name suffix based on client credentials. The method is needed to be able
+    to return metadata w/o a suffix to the client.
+    This method is meant to be overwritten by derived classes.
+
+    :param credDict: client credentials
+    :return: the suffix. And empty string for a base class.
+    """
+
+    return ''
   def addMetadataField(self, pname, ptype, credDict):
     """ Add a new metadata parameter to the Metadata Database.
         pname - parameter name, ptype - parameter type in the MySQL notation
+        Modified to use fully qualified metadata names.
     """
-
+    # existing pnames are fully qualified, so
+    fqPname = self.getMetaName(pname, credDict)
     result = self.db.fmeta.getFileMetadataFields(credDict)
     if not result['OK']:
       return result
-    if pname in result['Value'].keys():
+    if fqPname in result['Value'].keys():
```

Comment:
Please avoid the `.keys()` methods (not python 3 friendly)

Reply:
changed it to read: 
`if fqPname in result['Value']:`

I don't see why the original code (present all over the places in the same file would be not Python3 friendly. The `in` operator works file in Python 3 even thought `.keys()` does not return a list.